src/java/org/apache/cassandra/index/sasi/conf/view/PrefixTermTree.java

```diff
@@ -30,69 +30,60 @@
 import org.apache.cassandra.index.sasi.utils.trie.Trie;
 import org.apache.cassandra.db.marshal.AbstractType;
 import org.apache.cassandra.utils.Interval;
-import org.apache.cassandra.utils.IntervalTree;
+import org.apache.cassandra.utils.IntervalList;
 
 import com.google.common.collect.Sets;
 
 /**
- * This class is an extension over RangeTermTree for string terms,
- * it is required because interval tree can't handle matching if search is on the
- * prefix of min/max of the range, so for ascii/utf8 fields we build an additional
- * prefix trie (including both min/max terms of the index) and do union of the results
- * of the prefix tree search and results from the interval tree lookup.
+ * This class is an extension over RangeTermTree for string terms, it is
+ * required because interval tree can't handle matching if search is on the
+ * prefix of min/max of the range, so for ascii/utf8 fields we build an
+ * additional prefix trie (including both min/max terms of the index) and do
+ * union of the results of the prefix tree search and results from the interval
+ * tree lookup.
  */
-public class PrefixTermTree extends RangeTermTree
-{
+public class PrefixTermTree extends RangeTermTree {
     private final OnDiskIndexBuilder.Mode mode;
     private final Trie<ByteBuffer, Set<SSTableIndex>> trie;
 
-    public PrefixTermTree(ByteBuffer min, ByteBuffer max,
-                          Trie<ByteBuffer, Set<SSTableIndex>> trie,
-                          IntervalTree<Term, SSTableIndex, Interval<Term, SSTableIndex>> ranges,
-                          OnDiskIndexBuilder.Mode mode,
-                          AbstractType<?> comparator)
-    {
+    public PrefixTermTree(ByteBuffer min, ByteBuffer max, Trie<ByteBuffer, Set<SSTableIndex>> trie,
+            IntervalList<Term, SSTableIndex, Interval<Term, SSTableIndex>> ranges, OnDiskIndexBuilder.Mode mode,
+            AbstractType<?> comparator) {
         super(min, max, ranges, comparator);
 
         this.mode = mode;
         this.trie = trie;
     }
 
-    public Set<SSTableIndex> search(Expression e)
-    {
-        Map<ByteBuffer, Set<SSTableIndex>> indexes = (e == null || e.lower == null || mode == OnDiskIndexBuilder.Mode.CONTAINS)
-                                                        ? trie : trie.prefixMap(e.lower.value);
+    public Set<SSTableIndex> search(Expression e) {
+        Map<ByteBuffer, Set<SSTableIndex>> indexes = (e == null || e.lower == null
+                || mode == OnDiskIndexBuilder.Mode.CONTAINS) ? trie : trie.prefixMap(e.lower.value);
 
         Set<SSTableIndex> view = new HashSet<>(indexes.size());
         indexes.values().forEach(view::addAll);
         return Sets.union(view, super.search(e));
     }
 
-    public static class Builder extends RangeTermTree.Builder
-    {
+    public static class Builder extends RangeTermTree.Builder {
```

Comment:
Our static analysis engine detects that your pull request includes a change that **class name shadows simple name of superclass**. According to [the best practice](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-shouldn-t-shadow-simple-name-of-superclass-nm-same-simple-name-as-superclass), it can be exceptionally confusing, creating many situations where you have to look at import statements to resolve references, and creating many opportunities to accidentally define methods that don't override the methods in their superclasses.